pkg/apis/pipeline/v1alpha1/pipeline_defaults_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestPipeline_SetDefaults(t *testing.T) {
+	cases := []struct {
+		desc string
+		p    *v1alpha1.Pipeline
+		want *v1alpha1.Pipeline
+	}{{
+		desc: "empty pipeline",
+		p:    &v1alpha1.Pipeline{},
+		want: &v1alpha1.Pipeline{},
+	}}
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			ctx := context.Background()
+			tc.p.SetDefaults(ctx)
+			if diff := cmp.Diff(tc.want, tc.p); diff != "" {
+				t.Errorf("Mismatch of Pipeline: %s", diff)
```

Comment:
Done, added tc.desc as part of the Errorf along with the diff to make it little more easier instead of reading through `_` for example:

```
TestPipelineSpec_SetDefaults/pipeline_task_-_default_task_kind_must_be_Task: pipeline_defaults_test.go:134: Mismatch of pipelineSpec after setting defaults: pipeline task - default task kind must be Task: (-want, +got):   &v1beta1.PipelineSpec{
...
```